src/traqball.js

```diff
@@ -278,20 +278,24 @@
 
             curTime = new Date().getTime();
         }
-    
-        function eulerAngles(R) {
-          // r0: r0,0 r1: r0,1 r2: r0,2 r3: r0,3
-          // r4: r1,0 r5: r1,1 r6: r1,2 r7: r1,3
-          // r8: r2,0 r9: r2,1 r10: r2,2 r11: r2,3
 
-          // these formulas taken from
-          // https://www.learnopencv.com/rotation-matrix-to-euler-angles/
+        function eulerAngles(R) {
+          // Get camera vector
+          var M = new THREE.Matrix4();
+          M.fromArray(R);
+          M = M.getInverse(M);
+          euler = new THREE.Euler();
 
-          var sy = Math.sqrt(R[0] * R[0] +  R[4] * R[4]);
+          // Get rotation vector (to match initial transform?)
+          rot = new THREE.Matrix4();
+          rot.makeRotationZ(Math.PI/2);
+          M.premultiply(rot);
 
-          euler.roll  = radToDeg(Math.atan2(R[9], R[10]));
-          euler.pitch = radToDeg(Math.atan2(-R[8], sy));
-          euler.yaw   = radToDeg(Math.atan2(R[4], R[0]));
+          // This gives us the desired values
+          euler.setFromRotationMatrix(M, 'XYZ');
+          euler.roll = radToDeg(euler.x);
```

Comment:
These are getting set on a threejs Euler() object, but I should probably clean this up to use a private var for that so euler stays as it was before.